map/traffic_manager.cpp

```diff
@@ -327,7 +318,12 @@ void TrafficManager::OnTrafficDataResponse(traffic::TrafficInfo && info)
 
   // Update traffic colors for drape.
   df::TrafficSegmentsColoring coloring;
-  coloring[info.GetMwmId()] = info.GetColoring();
+  auto & mwmColoring = coloring[info.GetMwmId()];
```

Comment:
It looks like coloring filling and drape updating can be performed out of m_mutex lock scope. Isn't it?